djes/search.py

```diff
@@ -52,11 +52,27 @@ def hits(self):
 class LazySearch(Search):
     """This extends the base Search object, allowing for Django-like lazy execution"""
 
+    def __init__(self, *args, **kwargs):
+        super(LazySearch, self).__init__(*args, **kwargs)
+        self.position = kwargs.get("from", 0)
+
     def __len__(self):
         if 'size' in self._extra:
             return min(self._extra['size'], self.count())
         return self.count()
 
+    def __next__(self):
+        return self.next()
+
+    def next(self):
+        # Tracks index on the search object so we can iterate across multiple queries.
+        start, end = self.position, self.position + 1
+        self.position += 1
+        try:
+            return self[start:end][0]
```

Comment:
This is one of the issues I have brought up before. We are currently using `full` in circumstances where we would otherwise be retrieving from es. So if you iterate over a `LazySearch` object as is, you will only get 10 objects maximum. If you iterate over `LazySearch.full()` you will get everything but it defeats the purpose of using elasticsearch at all.